mp/src/game/server/neo/neo_player.cpp

```diff
@@ -2468,11 +2468,42 @@ int	CNEO_Player::OnTakeDamage_Alive(const CTakeDamageInfo& info)
 				++iDamage;
 				flDmgAccumlator -= 1.0f;
 			}
+
+			// Mirror team-damage
+			const bool bIsTeamDmg = (attackerIdx != entindex() && attacker->GetTeamNumber() == GetTeamNumber());
+			const float flMirrorMult = NEORules()->MirrorDamageMultiplier();
+			if (flMirrorMult > 0.0f && bIsTeamDmg)
+			{
+				// Attacker own-team-inflicting - Mirror the damage
+				// NEO NOTE (nullsystem): DO NOT call OnTakeDamage directly as that'll crash the game, instead
+				// let the gamerule handle this situation by causing the attacker to suicide when health gets
+				// depliciated
+				const float flInflict = static_cast<float>(iDamage) * flMirrorMult;
+				attacker->OnTakeDamage_Alive(CTakeDamageInfo(GetContainingEntity(INDEXENT(0)),
+															 GetContainingEntity(INDEXENT(0)), flInflict, DMG_SLASH));
```

Comment:
Nitpick: somewhat faster to call `GetWorldEntity()` here instead of indexing 0 via engine call.